### DISCUSSION TITLE
This accessory requires a home hub to be used with the Home app.

### DISCUSSION THREAD

**Discussion by wwwakcan** [HomeSpan/HomeSpan#548] [General]
I get this error. How can I fix ?
HomeSpan Version : 1.7.1
ESP Version : 2.0.6
![IMG_1623](https://user-images.githubusercontent.com/11073291/230063342-05652958-9f8e-49ab-a05a-d1f4db3605c2.PNG)

my code : 

```cpp
#include "HomeSpan.h" 
#include "DEV_ProgButton.h"     
#include <WiFi.h>

const char* ssid     = "XXXX";
const char* password = "XXXXXXXX";


void setup() {

  Serial.begin(115200);
  delay(10);

  homeSpan.begin(Category::Bridges,"VOBO HOME");
  homeSpan.setWifiCredentials(ssid, password);
  homeSpan.setLogLevel(2);
  homeSpan.setQRID("TEST");
  homeSpan.setPairingCode("19912199"); 
  new SpanAccessory();  
    new Service::AccessoryInformation();
      new Characteristic::Identify();

  new SpanAccessory();                                                          
    new Service::AccessoryInformation();
      new Characteristic::Identify(); 
      new Characteristic::Name("DEMO BUTON");

    new DEV_ProgButton(23,1); 
    new DEV_ProgButton(5,2); 
 
} // end of setup()

//////////////////////////////////////

void loop(){
  
  homeSpan.poll();
  
} // end of loop()

```

**Comment by HomeSpan** [comment]
Some HomeKit Accessories require background processing and automation routines.  Programmable buttons are one such type of Accessory.  HomeKit needs a Home Hub to be able to monitor for button presses and take an action when needed.  Home Hubs can be an Apple TV or a Home Pod.  HomeKit previously supported the use of a stay-at-home, always-on iPad as a Home Hub but I believe this was discontinued a year or so ago.

**Comment by wwwakcan** [comment]
I got it working and it works very well. Thankyou :)

**Reply by gaviniscool** [reply]
How did you do that? I got the same problem. Could you share your solution with us, thanks!
